chatbot/ai-studio.yaml

```diff
@@ -1,16 +1,36 @@
-application:
-  # type: language
-  # name: chatbot
-  # description: This is a LLM chatbot application
+applications:
+  type: language
+  name: chatbot
+  description: This is a LLM chatbot application
+  gitrepo: https://github.com/redhat-et/locallm
   containers:
-    - name: chatbot-example
-      contextdir: ai-application/builds
-      #image-name: quay.io/redhat-et/chatbot:latest
+    # podman run --rm -d -p 7860:7860 quay.io/sallyom/model-services:llama2-7b-chatbot-arm
     - name: chatbot-model-arm
-      contextdir: model-services/builds/arm
+      contextdir: .
+      containerfile: chatbot/model_service/builds/base/Containerfile
       arch: arm64
-      model-service: true 
+      model-service: true
+      #image-name: quay.io/sallyom/model-services:llama2-7b-chatbot-arm
     - name: chatbot-model-x86
-      contextdir: model-services/builds/x86
+      contextdir: .
+      containerfile: chatbot/model_service/builds/base/Containerfile
+      arch: amd64
+      model-service: true 
+      #image-name: quay.io/sallyom/model-services:llama2-7b-chatbot
+    - name: chatbot-model-x86-cuda
+      contextdir: .
+      containerfile: chatbot/model_service/builds/cuda/Containerfile
       arch: amd64
```

Comment:
This brings up the need for more thought around this configuration file - how about a new field like `backend` so like 
```
- name: chatbot-model-x86-cuda
  contextdir: .
  containerfile: chatbot/model_service/builds/x86-cuda/Containerfile
  arch: amd64
  backend: cuda (or llama, stablediffusion, etc,etc) see: https://github.com/go-skynet/model-gallery/tree/main for different examples. 
  ```
  I'll add `backend` for now.

Reply:
the `backend` will only be relevant for those images where `model-service: true` I suppose

Reply:
I don't think "cuda" and "llama" are in the same category. Right now the only "backend" (i've been using the term framework) we want to support is "llama" (which I assume is short for llama.cpp). "cuda" would be something like "accelerator" or "gpu_type" I think. Right now the only that we support (I think) are cuda and eventually Apple Metal.

Reply:
@MichaelClifford if a model service app supports llama, it would work with any llama models? If so, i think we could end up having different model services based on the "backend" used like 
```
model1:
   backend: 
     - llama
     - b
model2:
   backend:
     - c
     - ....
```
 and, consequently, i agree with using the term "gpu_type" or "accelerator" for cuda or apple metal.

Reply:
yeah so our current model service uses llama.cpp as the backend, which supports  any of the models listed [here](https://github.com/ggerganov/llama.cpp?tab=readme-ov-file#description).

If we wanted a different backend, I think that would require different image. So yes,I think we'd have different model services based on different backends. 

We probably need to build out a second "backend" to see how this would actually work though.

Reply:
should we add both fields? 
backend: llama
accelerator: cuda